visvis2/geometry/_mesh.py

```diff
@@ -0,0 +1,18 @@
+from python_shader import vec4, Array
+
+from ._base import Geometry
+
+
+# I think this is where this thing goes, right?
+
+
+class MeshGeometry(Geometry):
+    """ Geometry for a polygonial mesh.
+    """
+
+    # todo: we can use a definition like this to ensure that the shaders use the correct types
+    # todo: but then we must also validate that any binding arrays that get set match up
+    binding_def = {
+        "positions": (0, Array(vec4)),
+        "colors": (1, Array(vec4)),
+    }
```

Comment:
I think in THREE.JS you just take a regular Geometry and add the buffers + bindings as they are available. Not every mesh has a color buffer for example. Example

```js
var geometry = new THREE.BufferGeometry();
var vertices = new Float32Array( [
	-1.0, -1.0,  1.0,
	 1.0, -1.0,  1.0,
	 1.0,  1.0,  1.0,
	 1.0,  1.0,  1.0,
	-1.0,  1.0,  1.0,
	-1.0, -1.0,  1.0
] );
geometry.setAttribute( 'position', new THREE.BufferAttribute( vertices, 3 ) );
var material = new THREE.MeshBasicMaterial( { color: 0xff0000 } );
var mesh = new THREE.Mesh( geometry, material );
```

`geometry.setAttribute` defines the binding and supplies the buffer in one call here and avoids the need for a `MeshGeometry` subclass.